services/datamanager/datasync/upload_data_capture_file.go

```diff
@@ -20,22 +26,94 @@ func uploadDataCaptureFile(ctx context.Context, client v1.DataSyncServiceClient,
 		return nil
 	}
 
-	ur := &v1.DataCaptureUploadRequest{
-		Metadata: &v1.UploadMetadata{
-			PartId:           partID,
-			ComponentType:    md.GetComponentType(),
-			ComponentName:    md.GetComponentName(),
-			MethodName:       md.GetMethodName(),
-			Type:             md.GetType(),
-			MethodParameters: md.GetMethodParameters(),
-			FileExtension:    md.GetFileExtension(),
-			Tags:             md.GetTags(),
-		},
-		SensorContents: sensorData,
+	uploadMD := &v1.UploadMetadata{
+		PartId:           partID,
+		ComponentType:    md.GetComponentType(),
+		ComponentName:    md.GetComponentName(),
+		MethodName:       md.GetMethodName(),
+		Type:             md.GetType(),
+		MethodParameters: md.GetMethodParameters(),
+		FileExtension:    md.GetFileExtension(),
+		Tags:             md.GetTags(),
 	}
-	_, err = client.DataCaptureUpload(ctx, ur)
-	if err != nil {
-		return err
+
+	// If it's a large binary file, we need to upload it in chunks.
```

Comment:
Really great work making the streaming option readable and fit into the existing flow well by just checking the binary file size.